generate_wrappers.py

```diff
@@ -416,17 +428,57 @@ def generate_code(op, api_def, enum_store):
           input_names_and_types +
           attr_names_and_types +
           missing_types)]
+  body = ''
+  if len(return_name_and_types) == 0:
+    body = 'return #tfop({tfop_args})'.format(tfop_args=tfop_args)
+  elif len(return_name_and_types) >= 1:
+    # Example body with 1 return tensor:
+    # let ret: [TensorHandle<Int32>] = #tfop("ConcatOffset",
+    #   concatDim,
+    #   shape)
+    # return ret.0.map(Tensor.init)
+    #
+    # Example body with 2 return tensors:
+    # let ret: (loss: Tensor<T>, backprop: Tensor<T>) = #tfop("SoftmaxCrossEntropyWithLogits",
```

Comment:
The type signature is incorrect here, and the code example in this comment won't work. 

1. Tuple elements should be handles, not tensors.

2. If you add specific labels to the tuple type signature (like in your example `: (loss: Tensor<T>, backprop: Tensor<T>)`), then you'll need to access elements of `ret` via `ret.loss` and `ret.backprop`. Your examples below (`ret.0`, etc) won't work. So don't add labels to the type signature - this makes the compiler erase the label information so that you won't need labels to access them (you can then just use `ret.0`, `ret.1`, etc).

Reply:
Fixed this stale comment. Thanks for the catch!

The real code from RawOpsGenerated.swift (part of this PR) is:
```swift
@inlinable @inline(__always)
public static func softmaxCrossEntropyWithLogits<T: BinaryFloatingPoint>(
  features: Tensor<T>,
  labels: Tensor<T>
) -> (loss: Tensor<T>, backprop: Tensor<T>) {
  let ret: (loss: TensorHandle<T>, backprop: TensorHandle<T>) = #tfop("SoftmaxCrossEntropyWithLogits",
    features,
    labels,
    T: T.self)
  return (Tensor(handle: ret.0), Tensor(handle: ret.1))
}
```

using `ret.0` works (I've tested this PR in the context of https://github.com/apple/swift/pull/17648#pullrequestreview-133425846), and I'd prefer that since it keeps the (generated) code simpler and more readable.

Reply:
This approach is really nice.  I think it makes perfect sense for the wrapper generator to produce TensorHandle directly from #tfop's.  With this approach, can we completely remove the goop in CSApply that is handling Tensor results?

Reply:
@lattner Does https://github.com/apple/swift/pull/17648 do what you have in mind?

Reply:
Ok, turns out index access also works for named tuple elements. One question though:

```swift
let ret: (loss: TensorHandle<T>, backprop: TensorHandle<T>)
```

Why are element labels in the tuple type signature necessary? They are not type-checked by the `#tfop` type checker, and the are making the generated code *less* simple and *less* readable because they are unnecessarily long and you aren't using element labels to access elements anyway.

If you remove the tuple element labels in the type signature of `ret`
```swift
let ret: (TensorHandle<T>, TensorHandle<T>) = #tfop("SoftmaxCrossEntropyWithLogits",
  features,
  labels,
  T: T.self)
return (Tensor(handle: ret.0), Tensor(handle: ret.1))
```

This is arguably shorter and much simpler.

Reply:
I agree with removing tuple element labels from `#tfop` for simplicity. Labels should be preserved on the actual wrapper `func`.

Reply:
Done.